train.py

```diff
@@ -121,8 +121,8 @@ def train(hyp, opt, device, tb_writer=None, wandb=None):
     # plot_lr_scheduler(optimizer, scheduler, epochs)
 
     # Logging
-    if wandb and wandb.run is None:
-        wandb_run = wandb.init(config=opt, resume="allow",
+    if wandb:
```

Comment:
Checking `wandb.run is None` is done to make sure the user hasn't already initialized a wandb run manually. Consider a scenario where a user uses yolo a sub-part in a larger pipeline and calls `wandb.init` manually earlier in the pipeline, and now if we remove this check, wandb will again re-initialize a new run, which is not desirable. 
Keeping this check provides more power to the user to import this module into other projects without breaking the logging operation.

Reply:
@AyushExel you are right, the setattr() method does not return anything, so config was not being set (!). I've fixed that and restored the if statement.

I've also started pushing updates to image logging here in the same PR. My main idea is to wait until training finishes to show overlays and results, but to present training Mosaics right away so users can verify their labels are correct. 

I've also added all extra imagery generated in local logging to W&B now.

Reply:
@glenn-jocher  Looks good. Ideally, for the results plot you can also log the matplotlib plt object which would be interactive in dashboard allowing you to hover over the graph and get insights. But going that way will require more modification in plotting helper function. This looks great too. I tested it and everything is working good. Feel free to merge this whenever you're ready